### ISSUE TITLE
S3 wait block: NaN 

### ISSUE THREAD

**Issue by kgracey** [parallaxinc/BlocklyProp#516]
@dcarrier-parallax the wait block seems to produce "not a number" type of code errors. For example:
![2016-08-12_14-16-26](https://cloud.githubusercontent.com/assets/9534148/17636913/22a3c56e-6095-11e6-9efb-764bc1ee18be.png)

Creates the following:
```
CON
  _clkmode = xtal1 + pll16x
  _xinfreq = 5_000_000

VAR

OBJ
  scribbler    : "Block_Wrapper"

PUB Setup
  Scribbler.Start
  Start


PUB Start

  Scribbler.SetLED(Scribbler#LEFT, Scribbler#COLOR_00ff00)
  Scribbler.SetLED(Scribbler#CENTER, Scribbler#COLOR_00ff00)
  Scribbler.SetLED(Scribbler#RIGHT, Scribbler#COLOR_00ff00)
  waitcnt(clkfreq / 1 * NaN + 1000 + cnt) {TODO: catch short wait periods}
  Scribbler.SetLED(Scribbler#LEFT, Scribbler#COLOR_ffff00)
  Scribbler.SetLED(Scribbler#CENTER, Scribbler#COLOR_ffff00)
  Scribbler.SetLED(Scribbler#RIGHT, Scribbler#COLOR_ffff00)
  waitcnt(clkfreq / 1 * NaN + 1000 + cnt) {TODO: catch short wait periods}
  Scribbler.SetLED(Scribbler#LEFT, Scribbler#COLOR_ff0000)
  Scribbler.SetLED(Scribbler#CENTER, Scribbler#COLOR_ff0000)
  Scribbler.SetLED(Scribbler#RIGHT, Scribbler#COLOR_ff0000)
```
Is it treating the value as a string or character?

**Comment by dcarrier-parallax** [comment]
Fixed before code committed in commit #acf37ad

**Comment by wstodghill** [comment]
@kgracey and @dcarrier-parallax - In the dropdown menu - the following error occurs (at least for me - running Google Chrome or Explorer in Windows. See screen capture.

![wait block error](https://cloud.githubusercontent.com/assets/8891440/17641291/e9140148-60de-11e6-9bb3-b48da48d222e.JPG)

I propose these changes (which I think are technically more accurate too)

![wait block improved](https://cloud.githubusercontent.com/assets/8891440/17641296/0bb7c72a-60df-11e6-89a4-7ca3afa61e6d.JPG)

The block factory link is https://blockly-
demo.appspot.com/static/demos/blockfactory/index.html#f8baq3
